### PR TITLE
Lab12

### PR DESCRIPTION
**Pull request by Andrew174194**
Pull request for lab 12
by Andrey Khoroshavin (kill team, CS-01 academic group)

### CODE REVIEW

File: app_python/README.md [hunk 1]
```diff
@@ -52,6 +52,10 @@ Workflow steps:
 4) Unit tests
 5) Login to Docker Hub & push new image (pay attention to set `DOCKER_HUB_USERNAME` and `DOCKER_HUB_ACCESS_TOKEN` secrets)
 
+## /visits page
+Every time one visits the `/` index page, the new record in `/app/visits/visits.txt` saves the returned time and IP address of client.  
+This file should be common for all processes, e.g. containers in cluster.
```
Comment (DOCtorActoAntohich) [inline app_python/README.md:57]: bamog

File: k8s/12.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# File is in place, let's read it
+$ kubectl exec msc-time-py-54f755498c-g22ww -- cat config.json
+{
+    "version": "Hello World",
+    "history_text": "Records of time accesses"
+}
+```
+
+That's it! 😄🔫
```
Comment (DOCtorActoAntohich) [inline k8s/12.md]: ded

File: app_python/main.py [hunk 3]
```diff
@@ -16,10 +17,33 @@
...
+    with open("/app/visits/visits.txt", "a+") as fo:
+        fo.write(log_str)
+
+@bp.route('/visits')
+def visits():
+    web_content = "History of visits at timestamps and requester's IP:<br>"
+
+    try:
+        with open("/app/visits/visits.txt", "r") as fo:   
+            file_text = fo.read().replace('\n', '<br>')
```
Comment (DOCtorActoAntohich) [inline app_python/main.py:39]: funny, tho not sure if it's really necessary to `.replace` every time if you can just write `<br>` to files 

but :trollface:  if it works don't touch it

**Review by DOCtorActoAntohich — APPROVED** [review]
gud job

File: app_python/main.py (hunk 3 above)
Comment (Andrew174194) [reply app_python/main.py:39]: `Log file be like:<br>
Yes<br>`

Comment (ntdesmond) [reply app_python/main.py:39]: log file goes brrrrrrrrrrr

**Review by ntdesmond — APPROVED** [review]
win

**Pull request closed without merging**